subprojects/plugins/src/integTest/groovy/org/gradle/api/tasks/JavaExecIntegrationTest.groovy

```diff
@@ -99,4 +108,73 @@ class JavaExecIntegrationTest extends AbstractIntegrationSpec {
         then:
         ":run" in nonSkippedTasks
     }
+
+    def "arguments can be passed by using argument providers"() {
+        given:
+        def inputFile = file("input.txt")
+        def outputFile = file("out.txt")
+        buildFile << """
+            import org.gradle.process.CommandLineArgumentProvider
+
+            class MyApplicationJvmArguments implements CommandLineArgumentProvider {
+                @InputFile
+                @PathSensitive(PathSensitivity.NONE)
+                File inputFile
+            
+                @Override
+                Iterable<String> asArguments() {
+                    return ["-Dinput.file=\${inputFile.absolutePath}"]
+                }            
+            }
+            
+            class MyApplicationCommandLineArguments implements CommandLineArgumentProvider {
+                @OutputFile
+                File outputFile
+
+                @Override
+                Iterable<String> asArguments() {
+                    return [outputFile.absolutePath]
+                }            
+            }
+            
+            run.jvmArgProviders << new MyApplicationJvmArguments(inputFile: new File(project.property('inputFile')))
+            
+            run.argProviders << new MyApplicationCommandLineArguments(outputFile: new File(project.property('outputFile')))
+             
+        """
+        inputFile.text = "first"
+        mainJavaFile.text = mainClass("""
+            try {
+                String location = System.getProperty("input.file");
+                BufferedReader reader = new BufferedReader(new FileReader(location));
+                String input = reader.readLine();
+                reader.close();
+                FileWriter out = new FileWriter(args[args.length - 1], false);
+                out.write(input);
+                out.close();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            
+        """)
+
+        when:
+        run "run", "-PinputFile=${inputFile.absolutePath}", "-PoutputFile=${outputFile.absolutePath}"
+        then:
+        ":run" in nonSkippedTasks
```

Comment:
Why not use `executedAndNotSkipped`?

Reply:
Changed to `executedAndNotSkipped` for all occurrences in the test file.